app/Components/Sudoku Board/SudokuBoard.tsx

```diff
@@ -284,68 +289,85 @@ const SudokuBoard = (props: SudokuBoardProps) => {
   };
 
   const renderCell = (cell: CellProps, r: number, c: number) => {
-    // row and column values are incorrect here.
-    // console.log("RENDER CELL", c, "COLUMN", r, "ROW", cell)
-    const {
-      isHighlightIdenticalValues,
-      isHighlightBox,
-      isHighlightRow,
-      isHighlightColumn,
-    } = React.useContext(PreferencesContext);
-
-    let selected = sudokuBoard.selectedCell;
-    let isSelected = false;
-    let conflict = isConflict(r, c, cell);
-    let peer = false;
-    let box = false;
-    let row = false;
-    let column = false;
-    let sameValue = false;
-    if (selected != null) {
-      isSelected = c === selected.c && r === selected.r;
-      box = isCurrentCellAndSelectedCellInSameBox({ r: r, c: c }, selected);
-      row = isCurrentCellAndSelectedCellInSameRow({ r: r, c: c }, selected);
-      column = isCurrentCellAndSelectedCellInSameColumn(
-        { r: r, c: c },
-        selected
-      );
-      peer =
-        !conflict &&
-        ((box && isHighlightBox) ||
-          (row && isHighlightRow) ||
-          (column && isHighlightColumn));
-
-      const currentSelectedCell: CellProps | null = getCurrentSelectedCell();
-      if (currentSelectedCell == null) {
-        return;
-      }
-      const selectedEntry = currentSelectedCell.entry;
-      let currentEntry = cell.entry;
-      sameValue =
-        !conflict &&
-        isHighlightIdenticalValues &&
-        selectedEntry === currentEntry &&
-        currentEntry != 0;
+    const selectedCell = sudokuBoard.selectedCell;
+    const isSelected =
+      selectedCell == null
+        ? false
+        : c === selectedCell.c && r === selectedCell.r;
+    const conflict = isConflict(r, c, cell);
+    const peer = selectedCell
+      ? isCurrentCellPeer(r, c, selectedCell as CellLocation)
+      : false;
+    const sameValue = selectedCell ? doesCurrentCellHaveSameValue(cell) : false;
+
+    let cellBackgroundColor;
+    if (conflict && !isSelected) {
+      cellBackgroundColor = NOT_SELECTED_CONFLICT_COLOR;
+    } else if (conflict && isSelected) {
+      cellBackgroundColor = SELECTED_CONFLICT_COLOR;
+    } else if (isSelected) {
+      cellBackgroundColor = SELECTED_COLOR;
+    } else if (sameValue) {
+      cellBackgroundColor = IDENTICAL_VALUE_COLOR;
+    } else if (peer) {
+      cellBackgroundColor = PEER_SELECTED_COLOR;
+    } else {
+      cellBackgroundColor = "white";
     }
 
     return (
       <Cell
         onClick={(r: number, c: number) => {
           toggleSelectCell(r, c);
         }}
-        sameValue={sameValue}
-        isSelected={isSelected}
-        isPeer={peer}
+        backgroundColor={cellBackgroundColor}
         type={cell.type}
         entry={cell.entry}
-        conflict={conflict}
         key={r + ":" + c}
         c={c}
         r={r}
       />
     );
   };
 
+  const doesCurrentCellHaveSameValue = (cell: CellProps) => {
```

Comment:
Maybe rename to something like hasSelectedValue and then add a comment above it to further clarify?

Reply:
Good idea, I think "cellHasSelectedValue" is a good name.

Reply:
I went with doesCellHaveSelectedValue